### PR TITLE
Refactor on-core unit-tests

### PR DESCRIPTION
**Pull request by jlongever**
- Remaining protocols tests
- Remove rabbitmq-server dependency.

@RackHD/corecommitters @keedya @zyoung51 @uppalk1 @tannoa2

### CODE REVIEW

**Comment by keedya** [comment]
:+1:

**Comment by jlongever** [comment]
test this please

File: spec/lib/protocol/waterline-spec.js [hunk 1]
```diff
@@ -4,26 +4,37 @@
...
+        var Subscription = helper.injector.get('Subscription');
+        testSubscription = new Subscription({},{});
+        sinon.stub(testSubscription);
+        sinon.stub(messenger);
     });
 
     helper.after();
 
     it('should publish a created event', function() {
+        messenger.publish.resolves(Promise.resolve());
```
Comment (yyscamper) [inline spec/lib/protocol/waterline-spec.js]: Is  this and following line intentionally? I don't see any assertion statement.

Comment (benbp) [inline spec/lib/protocol/waterline-spec.js]: Promise.resolve is redundant regardless. You can get away with `messenger.publish.resolves()`

Comment (jlongever) [inline spec/lib/protocol/waterline-spec.js]: @yyscamper nothing to assert?  This is the original code, only adding the publish() stub. 
@benbp just added it for the sake of clarity.

Comment (yyscamper) [inline spec/lib/protocol/waterline-spec.js]: yes, no assertion here. @jlongever , I just see you don't touch those code, if could, it is a good chance to add some assertion statement here. Of course, add it to TODO list and do it in another PR is OK for me.

Comment (yyscamper) [inline spec/lib/protocol/waterline-spec.js]: I just think if we don't do it this time, maybe no one will raise this up again

**Comment by yyscamper** [comment]
@jlongever , those houndci warning is positive, you'd better fix them.

File: spec/lib/protocol/waterline-spec.js (hunk 1 above)
Comment (jlongever) [inline spec/lib/protocol/waterline-spec.js]: @yyscamper I meant what do you want to assert here?

Comment (benbp) [inline spec/lib/protocol/waterline-spec.js]: @jlongever probably something like: `return expect(waterlineProtocol.publishRecord(collection, 'updated', { id: 1 })).to.be.fulfilled`

Comment (benbp) [inline spec/lib/protocol/waterline-spec.js]: Functionally it's the same since a promise rejection will fail the test, but this way the code makes it a little clearer what the intention of the test is.

**Comment by jlongever** [comment]
@benbp thanks.   Is this a show stopper?

**Comment by benbp** [comment]
@jlongever not on the expect statement thing, but the resolves() thing would be nice to fix, because it's actually returning a new promise object as the value to the promise. Right now it's of no consequence because we aren't testing the messenger.publish resolution value, but I could see it potentially leading to test failures down the road. If you don't mind just doing a search and replace over your changes?

Otherwise +1

**Comment by jlongever** [comment]
@benbp thanks.  updated.

**Comment by benbp** [comment]
:+1:

**Comment by jlongever** [comment]
test this please
